packages/nimble-components/src/table-column/number-text/models/manually-translated-unit-scale-formatter.ts

```diff
@@ -0,0 +1,62 @@
+import { ManuallyTranslatedScaledUnit } from './manually-translated-scaled-unit';
+import type { ScaledUnit } from './scaled-unit';
+import type { UnitPrefix } from './unit-prefix';
+import type { UnitTranslation } from './unit-translation';
+import { UnitScaleFormatter } from './unit-scale-formatter';
+
+/**
+ * A formatter for a unit scale that is not supported by Intl.NumberFormat and has translations built into Nimble
+ */
+export abstract class ManuallyTranslatedUnitScaleFormatter extends UnitScaleFormatter {
+    private readonly unitTranslations: Map<string, UnitTranslation>;
+    private readonly supportedPrefixes: UnitPrefix[];
+
+    public constructor(
+        lang: string,
+        formatterOptions: Intl.NumberFormatOptions
+    ) {
+        super(lang, formatterOptions);
+        this.unitTranslations = this.getUnitTranslations();
+        this.supportedPrefixes = this.getSupportedPrefixes();
+        if (!this.unitTranslations.get('en')) {
+            throw new Error('English translations must exist');
+        }
+    }
+
+    protected override getSupportedUnits(
+        lang: string,
```

Comment:
> I'm trying to figure out if the lang parameters in this class hierarchy should actually be called locale instead

If we get technical its a full bcp47 descriptor, which is referred to as a language-tag and is more than just script and region:
![image](https://github.com/ni/nimble/assets/1588923/854966a6-1e97-48a1-9cfd-3bdc2e96b547)

JS tends to use the term locale for the strings, HTML tends to use lang for them. Nimble codebase seems to use `lang`  (187) more often than locale (87).

Maybe since most of these objects are wrappers around the Intl.NumberFormat apis and that uses `locale` then we can stick to locale for the Formatters and UnitScale objects. But stuff that touches the elements and the lang DesignToken should use lang.

Reply:
> Taking a step back, what use cases do we want to claim support for today or might we want to support in the future?
>
>1. specifying different unit translations for different regions (e.g. en-us meter vs en-ca metre)
>2. specifying different value formatting for different regions (I can't think of an example but it seems plausible since other values like dates are region-specific)
>3. falling back to a default translation for a language when the client requests a locale for that language but a different region (e.g. making sure fr-ca resolves to our fr or fr-fr translations and not our en translations)

We can support all three. I've made the necessary changes to support 1 (2 and 3 were already supported) and added test cases.